src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -1780,6 +1781,11 @@ public override object VisitField(FieldSymbol symbol, TypeCompilationState argum
                     NullableWalker.SnapshotManager? snapshotManager = null;
                     ImmutableDictionary<Symbol, Symbol>? remappedSymbols = null;
                     var compilation = bodyBinder.Compilation;
+                    constructorInitializer = methodBody is BoundConstructorMethodBody { Initializer: { } initializer }
```

Comment:
> It feels like it might be simpler to bind the implicit constructor initializer when the BoundConstructorMethodBody is created. In effect, make it so every BoundConstructorMethodBody for an instance constructor has an initializer.

I did think about this, but it will change the observable results in APIs such as IOperation. While I'm probably ok with making such a change, I'd want it to have dedicated tests and I don't think this PR is the place to make that change.

Reply:
> While I'm probably ok with making such a change, I'd want it to have dedicated tests and I don't think this PR is the place to make that change.

In that case, I would like to suggest to implement suggested refactoring in a separate PR, get it merged, and then come back to addressing the problem we are trying to address in this PR.